src/Backups/BackupImpl.cpp

```diff
@@ -907,7 +927,9 @@ void BackupImpl::finalizeWriting()
 void BackupImpl::increaseUncompressedSize(UInt64 file_size)
 {
     uncompressed_size += file_size;
+    processed_files_size += file_size;
     ++num_files;
+    ++num_processed_files;
```

Comment:
This is not a correct place to increase `num_processed_files` and `processed_files_size`. It's better to make a separate function `increaseProcessedSize()` and call it in the following places:
1. after increaseUncompressedSize() in `writeBackupMetadata()` and `readBackupMetadata()`
2. after `const auto & info = *info_opt;` in `BackupImpl::readFile()`
3. after `LOG_TRACE(log, "Writing backup for file {} from {}", file_name, from_file_name);` in `BackupImpl::writeFile()`

And this way `size_t BackupImpl::getNumProcessedFiles()` & `getProcessedFilesSize` will be simple

```
size_t BackupImpl::getNumProcessedFiles() const { std::lock_guard lock{mutex}; return num_processed_files; }
```

Reply:
@vitlibar could you please review again?